### PR TITLE
ci: Release Workflow

### PR DESCRIPTION
**Pull request by visch**
Ran this locally and it built properly. I didn't push up to PyPi or TestPyPi

Closes https://github.com/MeltanoLabs/tap-mysql/issues/27

### CODE REVIEW

File: pyproject.toml
Comment (edgarrmondragon) [inline pyproject.toml:1]: This is missing an update to the build-system table: https://github.com/mtkennerly/poetry-dynamic-versioning

File: .github/workflows/release_workflow.yml
Comment (edgarrmondragon) [inline .github/workflows/release_workflow.yml:1]: There's a better way to handle this using https://github.com/hynek/build-and-inspect-python-package, without explicitly installing Poetry and other tools.

It also could leverage PyPI trusted publishing.

See https://github.com/edgarrmondragon/tap-planetscaleapi/blob/main/.github/workflows/release.yaml for an example.

Comment (visch) [reply .github/workflows/release_workflow.yml:1]: Swapped it over to the same method you used for tap-planetsacleapi.

I tested building the package locally which did work (with a tag of v0.0.1) , the other steps look correct but I don't know how to test them locally easily

File: pyproject.toml
Comment (visch) [reply pyproject.toml:1]: fixed

File: pyproject.toml [hunk 1]
```diff
@@ -71,9 +93,14 @@ known-first-party = ["tap_mysql"]
 convention = "google"
 
 [build-system]
-requires = ["poetry-core==1.9.0"]
-build-backend = "poetry.core.masonry.api"
+requires = ["poetry-core==1.8.1", "poetry-dynamic-versioning"]
```
Comment (edgarrmondragon) [inline pyproject.toml:96]: Why downgrade poetry-core?

File: .github/workflows/release_workflow.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        fetch-depth: 0
+        ref: ${{ github.ref }}
+    - uses: hynek/build-and-inspect-python-package@v2
+
+  publish:
+    name: Publish to PyPI
+    if: startsWith(github.ref, 'refs/tags/')
+    runs-on: ubuntu-latest
+    environment:
+      name: publishing
```
Comment (edgarrmondragon) [inline .github/workflows/release_workflow.yml:22]:
```suggestion
      name: pypi
```

https://github.com/MeltanoLabs/tap-mysql/issues/26

File: pyproject.toml (hunk 1 above)
Comment (visch) [reply pyproject.toml:96]: No great reason I was testing, and this worked (it wasn't the cause of the issue I was having). Fixing

File: pyproject.toml [hunk 3]
```diff
@@ -71,9 +93,14 @@ known-first-party = ["tap_mysql"]
...
+build-backend = "poetry_dynamic_versioning.backend"
 
 [tool.poetry.scripts]
 # CLI declaration
 tap-mysql = 'tap_mysql.tap:TapMySQL.cli'
+
+[tool.poetry-dynamic-versioning]
+enable = true
+vcs = "git"
+style = "semver"
```
Comment (edgarrmondragon) [inline pyproject.toml:106]:
```suggestion
style = "pep440"
```

**Review by edgarrmondragon — APPROVED** [review]

**Pull request merged**